### ISSUE TITLE
rafcon library config relative path undefined behaviour

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#223]
I had a rafcon library config:

``` yaml
LIBRARY_PATHS:
    rmpl: rmpl
    test: test_rmpl
    links_and_nodes: links_and_nodes
TYPE: SM_CONFIG
```

When I loaded this config with rafcon started from the links and nodes manager the paths were resolved to /tmp/~relative_path~ and not relative to the config file path.
Changing the definition with ./ notation helped to resolve the issue:

``` yaml
LIBRARY_PATHS:
    rmpl: ./rmpl
    test: ./test_rmpl
    links_and_nodes: ./links_and_nodes
TYPE: SM_CONFIG
```

Originally created by @peter-lehner  at 2016-11-11 14:54:04+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
At the moment we used the config.yaml with absolute paths which were adapted by environment variables that can be used and are resolved while initiation of RAFCON. Examples for this are: `${HOME}/Desktop/rafcon_tutorial_library` or `${RAFCON_LIB_PATH}/generic`.
The support of relative paths and those always fix to includes config-path was not intended till now.

So the the defined behavior depends to your directory you are starting RAFCON from. Generally all tools in the world are interpreting your relative paths from where you start the tool from. The work around you propose is new to me and is maybe a yaml-file feature but was never intended as far as I know.

So this is a feature request to have a relative path support which is fixed to the config.yaml file loaded?! I think this could be useful.


Originally created by @Rbelder  at 2016-11-11 16:05:25+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Yes, I have always thought a relative path in the config.yaml to be relative to the absolute path of the config.yaml file. I had opened a feature request on this a long time ago and I even though @franzlst implemented this back then on my request, but I cannot find the issue right now.


Originally created by @peter-lehner  at 2016-11-11 16:33:07+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
OK @franzlst has made a interessting condition check in the library manager which is fixed. Original this issue was tackled end of April. At this time also "./" found its way into the code to be relative to the config.yaml handed as argument. Fix can be found in 617881738f2c. Your are welcome. :smile: 


Originally created by @Rbelder  at 2016-11-11 16:59:30+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Also works on Windows now :) 84fed82


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-11-14 07:46:56+00:00 (moved from RMC internal repository)
